src/compile/scale/assemble.ts

```diff
@@ -5,17 +5,21 @@ import {stringValue, vals} from '../../util';
 import {isDataRefDomain, isDataRefUnionedDomain, isFieldRefUnionDomain, isSignalRefDomain, VgDataRef, VgScale} from '../../vega.schema';
 import {Model} from '../model';
 import {isRawSelectionDomain, selectionScaleDomain} from '../selection/selection';
+import {mergeDomains} from './domain';
 
 
 export function assembleScale(model: Model): VgScale[] {
-    return vals(model.component.scales).reduce((scales: VgScale[], scaleComponent) => {
+    return vals(model.component.scales).reduce((scales, scaleComponent) => {
       if (scaleComponent.merged) {
         // Skipped merged scales
         return scales;
       }
 
       // We need to cast here as combine returns Partial<VgScale> by default.
-      const scale = scaleComponent.combine(['name', 'type', 'domain', 'domainRaw', 'range']) as VgScale;
+      const scaleComProps = scaleComponent.combine(['name', 'type', 'domainRaw', 'range']);
+      // FIXME: why do I need to delete this property?
+      delete scaleComProps.domains;
```

Comment:
I thought we agree to just make `domains` a part of the scale component directly? 

In any case, having a `// FIXME: why do I need to delete this property?` comment isn't good.